src/css-loader-core/loader.js

```diff
@@ -65,43 +66,61 @@ export default class FileSystemLoader {
 	fetch(_newPath, relativeTo, _trace) {
 		let newPath = _newPath.replace(/^["']|["']$/g, ""),
 			trace = _trace || String.fromCharCode(this.importNr++);
+		const useFileResolve = typeof this.fileResolve === "function";
 		return new Promise((resolve, reject) => {
-			let relativeDir = path.dirname(relativeTo),
-				rootRelativePath = path.resolve(relativeDir, newPath),
-				fileRelativePath = path.resolve(
-					path.resolve(this.root, relativeDir),
-					newPath
-				);
+			(useFileResolve
+				? this.fileResolve(newPath, relativeTo)
+				: Promise.resolve()
+			).then((fileResolvedPath) => {
+				if (fileResolvedPath && !path.isAbsolute(fileResolvedPath)) {
+					reject(
+						'The returned path from the "fileResolve" option must be absolute.'
+					);
+				}
+				let relativeDir = path.dirname(relativeTo),
```

Comment:
Let's define the variables using separate `let` statements to make them more explicit